sched/signal/sig_dispatch.c

```diff
@@ -424,7 +424,18 @@ int nxsig_tcbdispatch(FAR struct tcb_s *stcb, siginfo_t *info)
 
       if (stcb->task_state == TSTATE_WAIT_SEM)
         {
+#ifndef CONFIG_DISABLE_PTHREAD
+          if (info->si_signo == SIGCONDTIMEDOUT)
+            {
+              nxsem_wait_irq(stcb, ETIMEDOUT);
+            }
+          else
+            {
+              nxsem_wait_irq(stcb, EINTR);
```

Comment:
This looks like a hack trying to overcome a gap in design. I need some more time to think on it. Please hold on with merging of this PR.